compiler/resolution/functionResolution.cpp

```diff
@@ -5898,6 +5898,66 @@ static bool isUndecoratedClassNew(CallExpr* newExpr, Type* newType) {
   return false;
 }
 
+static void warnForThrowNotOwned(CallExpr* newExpr, Type* newType, Type* manager) {
+  // This function implements a warning that should be removed in 1.20
+  // in favor e.g. calling compilerError in chpl_fix_thrown_error.
+  INT_ASSERT(newExpr->parentSymbol);
+  Type* cType = canonicalClassType(newType);
+
+  if (isClass(cType) && isSubTypeOrInstantiation(cType, dtError)) {
+
+    bool unmanaged = false;
+    bool borrowed = false;
+    bool owned = false;
+    bool undecorated = false;
+
+    if (manager == dtOwned || isSubTypeOrInstantiation(newType, dtOwned))
+      owned = true;
+    else if (manager == dtUnmanaged || isUnmanagedClassType(newType))
+      unmanaged = true;
+    else if (manager == NULL)
+      undecorated = true;
+    else if (manager == dtBorrowed)
+      borrowed = true;
+
+    if (owned)
+      return; // throw owned Error is OK
+
+    SymExpr* checkSe = NULL;
+    if (CallExpr* parentCall = toCallExpr(newExpr->parentExpr))
+      if (parentCall->isPrimitive(PRIM_MOVE) ||
+          parentCall->isPrimitive(PRIM_ASSIGN))
+        if (SymExpr* lhsSe = toSymExpr(parentCall->get(1)))
+          checkSe = lhsSe;
+
+    if (checkSe) {
+      for_SymbolSymExprs(se, checkSe->symbol()) {
+        if (se == checkSe) {
+          // OK
+        } else if (CallExpr* parentCall = toCallExpr(se->parentExpr)) {
+          if (parentCall->isNamed("chpl_fix_thrown_error")) {
+            if (undecorated)
+              USR_WARN(parentCall,
+                       "please use 'throw new owned %s "
+                       "instead of 'throw new %s'",
+                       cType->symbol->name, cType->symbol->name);
+            else if (unmanaged)
+              USR_WARN(parentCall,
+                       "please throw owned %s instead of unmanaged %s",
+                       cType->symbol->name, cType->symbol->name);
+            else if (borrowed)
+              USR_FATAL(parentCall,
+                       "please throw owned %s instead borrowed %s",
```

Comment:
"instead of borrowed"

Reply:
Also, there's inconsistency here in the use of '.  The first message is missing a terminating ' for the owned section, and the second and third don't use ' at all.  I'd prefer they all use '.